lib/3scale/backend/job_fetcher.rb

```diff
@@ -64,35 +52,9 @@ def fetch
       def start(job_queue)
         loop do
           break if @shutdown
-
-          if job_queue.size >= @max_pending_jobs
-            sleep @wait_before_fetching_more
-          else
-            begin
-              job = fetch
-            rescue RedisConnectionError => e
-              # If there has been a connection error or a timeout we wait a bit
-              # because normally, it will be a temporary problem.
-              # In the future, we might want to put a limit in the total number
-              # of attempts or implement exponential backoff retry times.
-              Worker.logger.notify(e)
-              sleep(1)
-
-              # Re-instantiate Redis instance. This is needed to recover from
-              # Errno::EPIPE, not sure if there are others.
-              @redis = Redis::Namespace.new(
-                WorkerAsync.const_get(:RESQUE_REDIS_NAMESPACE),
-                redis: QueueStorage.connection(Backend.environment, Backend.configuration)
-              )
-
-             # If there is a different kind of error, it's probably a
-             # programming error. Like sending an invalid blpop command to
-             # Redis. In that case, let the worker crash.
-            end
-            job_queue << job if job
-          end
+          job = fetch
+          job_queue << job if job
```

Comment:
leave reconnection to how it is handled by the default client. I dunno if reconnection is done there. But it would be the same as the job processors. It makes no sense if job processing fails to keep running fetcher anyway.

Reply:
What you say sounds logic. In the case of a connection error, what the client does by default is crashing, AFAIK. Then OCP starts another container to replace the crashed one.